packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -160,9 +162,39 @@ export class DocumentDeltaConnection
 				0x20b /* "mismatch" */,
 			);
 			if (!this.trackedListeners.has(event)) {
-				this.addTrackedListener(event, (...args: any[]) => {
-					this.emit(event, ...args);
-				});
+				if (event === "pong") {
+					// Garbage so we don't subscribe to any event coming from socket
+					this.trackedListeners.set("pong", () => {});
+
+					const sendPingLoop = () => {
+						const start = Date.now();
+
+						// Log if latency is longer than 1 min
+						this.socket.volatile?.emit("ping", () => {
+							const latency = Date.now() - start;
+							if (latency > 1000 * 60) {
+								this.mc.logger.sendErrorEvent({
+									eventName: "LatencyTooLong",
+									driverVersion,
+									duration: latency,
```

Comment:
Should this logic be not here, but rather in ContainerRuntime's listener that reports data to telemetry?